test-cypress/steps/import-steps.js

```diff
@@ -0,0 +1,225 @@
+/**
+ * Reusable functions for interacting with the import page.
+ */
+class ImportSteps {
```

Comment:
I don't like this class very much. It encompasses and mixes logic which is related to the precondition, action and verification steps. Although the steps are very granular and simple, they break the BDD approach which I think we should follow. Also those steps doesn't say much about what's tested and what's not actaully because the assertions are mixed within the actions. Not to mention that with this approach many verifications are now executed with every test in the spec which might not be necessary.

Reply:
Yes, most of them were commands, I gave them scope with this class.

I find it difficult to separate logic, assertion & wait in Cypress. Especially because assertion & waiting are under the same hood. Cypress bundles default assertions in most of its actions/commands!
I don't see how an action could wait for some condition without an assertion.

So, what do you suggest? Split actions from assertions in separate methods or split and move elsewhere?

Reply:
An assertion usually do wait of course and I don't mean we have to separate them. 
> Cypress bundles default assertions in most of its actions/commands! I don't see how an action could wait for some condition without an assertion.

And I got confused by this comment. I don't know what you mean, but cypress doesn't bundle assertions with actions. It performs checks by following some rules before executing any action in order to verify the state of the elements so to be able to report proper errors if need. See this for example: https://docs.cypress.io/guides/core-concepts/interacting-with-elements.html#Visibility
Given this, its not necessary to do assertion for visibility before clicking a button for example because cypress will blow with appropriate error message in either case.
I could comment case by case in the file below what I mean if you want. 
Also I'm OK if you disregard this comment for now because we could improve these on later stages.

Reply:
and some example tests from cypress repos :)
 https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/blogs__application-actions/cypress/integration/spec.js

Reply:
I will disregard it until you give examples how the code should be reorganised, improved etc

Reply:
OK. Will discuss it when I come back.